### PR TITLE
.Net: Plans using FunctionRunner + RequestSettings

### PR DESCRIPTION
**Pull request by RogerBarreto**
### Motivation and Context

Plans now fully utilize Kernel.RunAsync thru FunctionRunner interface, this will allow Plan steps to trigger Pre/Post Hooks defined in the Kernel that the Plan was ran.

Option to override the RequestSettings in the Kernel.RunAsync call. 
This is was required while running Plan steps that by default will use the current Plan settings.

One more step towards using RunAsync as the main entry point + Pre/Post hooks phase 2.

Resolves Partially #2324 

### Description

- Kernel.RunAsync optionally accepts `AIRequestSettings`
- PlanSteps now run in the Kernel and can trigger Pre/Post Hooks 

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/SemanticKernel.Abstractions/IKernel.cs [hunk 1]
```diff
@@ -52,12 +53,14 @@ public interface IKernel
...
     /// <param name="variables">Input to process</param>
+    /// <param name="requestSettings">Request settings to use in the functions call</param>
     /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
     /// <param name="pipeline">List of functions</param>
     /// <returns>Result of the function composition</returns>
     Task<KernelResult> RunAsync(
         ContextVariables variables,
-        CancellationToken cancellationToken,
+        AIRequestSettings? requestSettings = null,
+        CancellationToken cancellationToken = default,
```
Comment (markwallace-microsoft) [inline dotnet/src/SemanticKernel.Abstractions/IKernel.cs:63]: Does this need to be tagged as a breaking change?

Or, can we add a method to `KernelExtensions` that matches the old signature?

File: dotnet/src/SemanticKernel.Abstractions/Orchestration/IFunctionRunner.cs [hunk 2]
```diff
@@ -15,11 +16,13 @@ public interface IFunctionRunner
...
     /// <param name="skFunction">Target function to run</param>
     /// <param name="variables">Input to process</param>
+    /// <param name="requestSettings">Request settings to override</param>
     /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
     /// <returns>Result of the function composition</returns>
     Task<FunctionResult> RunAsync(
         ISKFunction skFunction,
         ContextVariables? variables = null,
+        AIRequestSettings? requestSettings = null,
         CancellationToken cancellationToken = default);
```
Comment (markwallace-microsoft) [inline dotnet/src/SemanticKernel.Abstractions/Orchestration/IFunctionRunner.cs:26]: Not expecting anyone will have implemented this but if they have this makes the change a breaking one, so we will need to add that tag.

File: dotnet/src/SemanticKernel.Core/Planning/Plan.cs [hunk 3]
```diff
@@ -336,10 +336,19 @@ public FunctionView Describe()
...
+                functionVariables,
+                requestSettings,
+                cancellationToken)
+                .ConfigureAwait(false);
+
+            /*result = await this.Function
                 .WithInstrumentation(context.LoggerFactory)
                 .InvokeAsync(functionContext, requestSettings, cancellationToken)
-                .ConfigureAwait(false);
+                .ConfigureAwait(false);*/
```
Comment (markwallace-microsoft) [inline dotnet/src/SemanticKernel.Core/Planning/Plan.cs:350]: Remove commented out code

**Review by SergeyMenshykh — APPROVED** [review]

File: dotnet/src/SemanticKernel.Abstractions/IKernel.cs (hunk 1 above)
Comment (RogerBarreto) [reply dotnet/src/SemanticKernel.Abstractions/IKernel.cs:63]: Added

File: dotnet/src/SemanticKernel.Abstractions/Orchestration/IFunctionRunner.cs (hunk 2 above)
Comment (RogerBarreto) [reply dotnet/src/SemanticKernel.Abstractions/Orchestration/IFunctionRunner.cs:26]: You are right, added an FunctionRunner extensions class to handle that.

**Review by dmytrostruk — COMMENTED** [review]
Let's discuss these changes before merging, if possible.

**Review by markwallace-microsoft — APPROVED** [review]

**Pull request merged**